tools/samtools.wdl

```diff
@@ -137,18 +168,26 @@ task flagstat {
     runtime {
         memory: "~{memory_gb} GB"
         disk: "~{disk_size_gb} GB"
-        docker: 'quay.io/biocontainers/samtools:1.16.1--h6899075_1'
+        docker: 'quay.io/biocontainers/samtools:1.17--h00cdaf9_0'
         maxRetries: max_retries
     }
 }
 
 task index {
     meta {
-        description: "This WDL task runs samtools index on the input BAM file."
+        description: "Creates a `.bai` BAM index for the input BAM"
+        outputs: {
+            bam_index: "A `.bai` BAM index associated with the input BAM. Filename will be `basename(bam) + '.bai'`."
```

Comment:
Why are we doing `basename(bam) + '.bai'` rather than `bam.bai`?

Reply:
To get rid of the path part of the name. I suppose that's an unnecessary clarification here. From a user POV `basename(bam) + '.bai'` is the same as `bam + '.bai'`.

Reply:
Nevermind. I read that as `basename(bam, ".bam") + '.bai'` even though that clearly isn't what is written. Hence my confusion. I despise the bare `.bai` suffix (as opposed to `.bam.bai`).